zyte_api/aio/client.py

```diff
@@ -43,6 +45,38 @@ def _post_func(session):
         return session.post
 
 
+class ExtractResult(Mapping):
+    """Result of a call to AsyncClient.extract.
+
+    It can be used as a dictionary to access the raw API response.
+
+    It also provides some helper properties for easier access to some of its
+    underlying data.
+    """
+
+    def __init__(self, api_response: dict):
+        self._api_response = api_response
+
+    def __getitem__(self, key):
+        return self._api_response[key]
+
+    def __iter__(self):
+        yield from self._api_response
+
+    def __len__(self):
+        return len(self._api_response)
+
+    @property
+    def http_response_body(self): -> bytes:
+        if hasattr(self, "_http_response_body"):
+            return self._http_response_body
+        base64_body = self._api_response.get("httpResponseBody", None)
+        if base64_body is None:
+            raise ValueError("API response has no httpResponseBody key.")
+        self._http_response_body = b64decode(base64_body)
+        return self._http_response_body
```

Comment:
We could later add `http_response_text`, which would return `str`, handling decoding.